evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -223,13 +233,36 @@ def get_feature_names(self):
                 # Follow sklearn naming convention but if name has been seen before
                 # then add an int to make it unique
                 proposed_name = self._make_name_unique(f"{col}_{category}", seen_before)
+                if (col, category) in self._binary_values_to_drop:
+                    encoded_features_to_drop.append(proposed_name)
 
                 unique_names.append(proposed_name)
                 unique_encoded_columns.append(proposed_name)
                 seen_before.add(proposed_name)
-            provenance[col] = unique_encoded_columns
+            self._features_to_drop.extend(encoded_features_to_drop)
```

Comment:
nit-pick: `encoded_features_to_drop` will only ever have at most one element right? I wonder if we can append directly to `_features_to_drop` as opposed to creating a new list and then calling extend of `_features_to_drop`?

Reply:
I think `encoded_features_to_drop` technically can have multiple values as long as multiple columns have binary categorical values that need to be encoded, but agree still that we can just append to `self._features_to_drop` rather than creating a new list.

Reply:
Bahaha I agree with both of you! The reason why I have it this way is because of `unique_encoded_columns_without_dropped` and `provenance`, lol. `provenance` is calculated for each column. We don't want to include the encoded features that we will drop. If I just directly append to `_features_to_drop`, I will still need to keep track of which categories for this column need to be dropped so I don't add to the unique_encoded_columns. If you guys have a suggestion to clean this up then I'd love to hear it, but that's why it's structured like this right now!